src/librustdoc/clean/mod.rs

```diff
@@ -1704,8 +1731,24 @@ pub(crate) fn clean_middle_ty<'tcx>(
                 })
                 .collect();
 
+            let mut late_bound_regions: Vec<_> = obj
+                .iter()
+                .flat_map(|pb| cx.tcx.collect_referenced_late_bound_regions(&pb))
```

Comment:
Concatenating + deduplicating pb.bound_vars() should do the right thing here.

Reply:
Ye, I thought about using `bound_vars` here as well after you recommended it to me. However, I'd need to collect the iterator into an `FxIndexSet` for deduplication and immediately after convert it into a `Vec` (since that's what the rest of the code expects) which requires reallocation I assume. That's why I dropped the idea but maybe it's not too bad actually.